tests/test_e2e.py

```diff
@@ -472,38 +516,51 @@ def test_b2c_acquire_token_by_auth_code(self):
                 # This won't work https://msidlab.com/api/user?usertype=b2c
             password="***"  # From https://aka.ms/GetLabUserSecret?Secret=msidlabb2c
         """
-        scopes = ["https://msidlabb2c.onmicrosoft.com/msaapp/user_impersonation"]
-        (self.app, ac, redirect_uri) = _get_app_and_auth_code(
-            "b876a048-55a5-4fc5-9403-f5d90cb1c852",
-            client_secret=self.get_lab_user_secret("MSIDLABB2C-MSAapp-AppSecret"),
-            authority=self._build_b2c_authority("B2C_1_SignInPolicy"),
-            port=3843,  # Lab defines 4 of them: [3843, 4584, 4843, 60000]
-            scopes=scopes,
-            )
-        result = self.app.acquire_token_by_authorization_code(
-            ac, scopes, redirect_uri=redirect_uri)
-        logger.debug(
-            "%s: cache = %s, id_token_claims = %s",
-            self.id(),
-            json.dumps(self.app.token_cache._cache, indent=4),
-            json.dumps(result.get("id_token_claims"), indent=4),
-            )
-        self.assertIn(
-            "access_token", result,
-            "{error}: {error_description}".format(
-                # Note: No interpolation here, cause error won't always present
-                error=result.get("error"),
-                error_description=result.get("error_description")))
-        self.assertCacheWorksForUser(result, scopes, username=None)
+        config = {"authority": self._build_b2c_authority("B2C_1_SignInPolicy"),
+                  "client_id": "b876a048-55a5-4fc5-9403-f5d90cb1c852",
+                  "scope": ["https://msidlabb2c.onmicrosoft.com/msaapp/user_impersonation"], "port": 3843}
+        self._test_acquire_token_by_auth_code(**config)
 
     def test_b2c_acquire_token_by_ropc(self):
-        self._test_username_password(
-            authority=self._build_b2c_authority("B2C_1_ROPC_Auth"),
-            client_id="e3b9ad76-9763-4827-b088-80c7a7888f79",
-            username="b2clocal@msidlabb2c.onmicrosoft.com",
-            password=self.get_lab_user_secret("msidlabb2c"),
-            scope=["https://msidlabb2c.onmicrosoft.com/msidlabb2capi/read"],
-            )
+        config = {"authority": self._build_b2c_authority("B2C_1_ROPC_Auth"),
```

Comment:
Given that the underlying `_test_username_password(...)` already accepts keyword arguments, don't you think the pre-existing calling pattern is actually more readable? Would suggest to revert changes to this test case. PS: The `**config` notation was only a "workaround" when the config was constructed by other helpers.